controllers/release/release_adapter_test.go

```diff
@@ -0,0 +1,423 @@
+
+		// required by test at line #273
+		applicationSnapshot.TypeMeta.Kind = "applicationSnapshot"
+		Expect(k8sClient.Update(ctx, applicationSnapshot)).Should(Succeed())
+	})
+
+	BeforeEach(func() {
+		release = &appstudiov1alpha1.Release{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-release-",
+				Namespace:    testNamespace,
+			},
+			Spec: appstudiov1alpha1.ReleaseSpec{
+				ApplicationSnapshot: applicationSnapshot.GetName(),
+				ReleaseLink:         releaseLink.GetName(),
+			},
+		}
+		Expect(k8sClient.Create(ctx, release)).Should(Succeed())
+
+		adapter = NewAdapter(release, ctrl.Log, k8sClient, ctx)
+		Expect(reflect.TypeOf(adapter)).To(Equal(reflect.TypeOf(&Adapter{})))
+	})
+
+	AfterEach(func() {
+		err := k8sClient.Delete(ctx, release)
+		Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
+	})
+
+	AfterAll(func() {
+		err := k8sClient.Delete(ctx, releaseLink)
+		Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
+
+		err = k8sClient.Delete(ctx, applicationSnapshot)
+		Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
+
+		Expect(k8sClient.Delete(ctx, releaseStrategy)).Should(Succeed())
+	})
+
+	It("can create a new Adapter instance", func() {
+		Expect(reflect.TypeOf(NewAdapter(release, ctrl.Log, k8sClient, ctx))).
+			To(Equal(reflect.TypeOf(&Adapter{})))
+	})
+
+	It("can make sure Finalizer is added to a Release object", func() {
+		result, err := adapter.EnsureFinalizerIsAdded()
+		Expect(!result.CancelRequest && err == nil).To(BeTrue())
+
+		Expect(release.ObjectMeta.GetFinalizers()[0]).
+			To(ContainSubstring("release-finalizer"))
+
+		// return success also when the finalizer is present already
+		//
+		// unf. the return type and error are the same so we can't
+		// differ the result and error from the previous call
+		result, err = adapter.EnsureFinalizerIsAdded()
+		Expect(!result.CancelRequest && err == nil).To(BeTrue())
+	})
+
+	It("can make sure a PipelineRun object exists for the instanced adapter", func() {
+		// some racing condition in EnvTest makes it throw an error even if the CancelRequest is false
+		// so I am adding a function to wait for it to succeed before continuing.
+		Eventually(func() bool {
+			result, err := adapter.EnsureReleasePipelineRunExists()
+			return (!result.CancelRequest && err == nil)
+		}).Should(BeTrue())
+
+		pipelineRun, _ := adapter.getReleasePipelineRun()
+		Expect(reflect.TypeOf(pipelineRun)).
+			To(Equal(reflect.TypeOf(&tektonv1beta1.PipelineRun{})))
+		Expect(pipelineRun).NotTo(BeNil())
+
+		// clean up
+		Expect(k8sClient.Delete(ctx, pipelineRun)).Should(Succeed())
+	})
+
+	It("can make sure that the status of a Release Pipeline is tracked", func() {
+		// same as above
+		Eventually(func() bool {
+			result, err := adapter.EnsureReleasePipelineRunExists()
+			return (!result.CancelRequest && err == nil)
+		}).Should(BeTrue())
+		pipelineRun, _ := adapter.getReleasePipelineRun()
+		Expect(pipelineRun).NotTo(BeNil())
```

Comment:
Got lost with the new commit, but the comments here https://github.com/redhat-appstudio/release-service/pull/71#discussion_r934476585 still apply to this line

Reply:
Still working on it. I am commenting before pushing it. Too long review.

Reply:
I think this one should be fine now. May you crosscheck?

Reply:
At time of writing, lines 158 and 170 are the ones I don't think we need. `Expect(pipelineRun).NotTo(BeNil())` we already check for it to be not nil in `getReleasePipelineRun()`